Automattic/jetpack
# WordPress.com Features: Update Calypso Locale Sync from wp-admin to Calypso

Fixes p58i-hkP-p2

## Proposed changes:

* Follow up to https://github.com/Automattic/jetpack/pull/37316
* Use `personal_options_update` hook instead of `insert_user_meta` and `update_option_WPLANG`
* `insert_user_meta` is called too frequently and can potentially cause "runaway" async jobs (Please see the P2 post above). This PR uses `personal_options_update` instead and adds additional "defensive" checks to prevent unnecessary API calls.

### Other information:


## Jetpack product discussion


Fixes p58i-hkP-p2

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






* The testing instructions on https://github.com/Automattic/jetpack/pull/37316 can be reused here.
* You can apply this PR to your Atomic site using the instruction here: PCYsg-Osp-p2
* You can place `error_logs` in the code to confirm `update_calypso_locale` isn't called unnecessarily, for example, by updating `/wp-admin/profile.php` but not changing the locale. Site logs are easily viewable at https://wordpress.com/site-logs/:siteSlug
* Note that the missing language pack installation job has been temporarily disabled in D151347-code. It will need to be re-enabled when we re-enable wp-admin to Calypso syncing.
